core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3253,3 +3253,18 @@ private[spark] class CircularBuffer(sizeInBytes: Int = 10240) extends java.io.Ou
     new String(nonCircularBuffer, StandardCharsets.UTF_8)
   }
 }
+
+object WebUrlUtils {
+  def getSparkWebUrl(context: SparkContext): Option[String] = {
```

Comment:
how is this different from `sc.uiWebUrl`?

Reply:
That’s a good question. It was originally extracted from `SparkILoop` as a `common` method
https://github.com/apache/spark/blob/f2d39b958deb9bae2cc39410144fe263dc96a0ef/repl/src/main/scala/org/apache/spark/repl/SparkILoop.scala#L47-L59

I think it depends on the following configuration:
https://spark.apache.org/docs/3.5.2/configuration.html
<img width="1105" alt="image" src="https://github.com/user-attachments/assets/cb1a1922-ec33-4c63-b359-60061897c798">

However `sc.uiWebUrl` depends on the following configuration:
https://github.com/apache/spark/blob/f2d39b958deb9bae2cc39410144fe263dc96a0ef/core/src/main/scala/org/apache/spark/ui/WebUI.scala#L175
https://github.com/apache/spark/blob/f2d39b958deb9bae2cc39410144fe263dc96a0ef/core/src/main/scala/org/apache/spark/ui/WebUI.scala#L58
https://github.com/apache/spark/blob/f2d39b958deb9bae2cc39410144fe263dc96a0ef/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1161-L1165

Reply:
I think we should align this logic in other places where `sc.uiWebUrl` is used, but of course, this is another PR. What I want to do next is. eg:
- spark-sql
https://github.com/apache/spark/blob/f2d39b958deb9bae2cc39410144fe263dc96a0ef/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala#L419